subsys/bluetooth/audio/cap_initiator.c

```diff
@@ -196,15 +205,98 @@ int bt_cap_initiator_broadcast_get_base(struct bt_cap_broadcast_source *source,
 #endif /* CONFIG_BT_AUDIO_BROADCAST_SOURCE */
 
 #if defined(CONFIG_BT_AUDIO_UNICAST_CLIENT)
-static const struct bt_uuid *cas_uuid = BT_UUID_CAS;
+enum {
+	CAP_UNICAST_PROCEDURE_STATE_ACTIVE,
+	CAP_UNICAST_PROCEDURE_STATE_ABORTED,
+
+	CAP_UNICAST_PROCEDURE_STATE_FLAG_NUM,
+} cap_unicast_procedure_state;
```

Comment:
Hmm, not at the moment. Used flags to use the atomic functions. The alternative would be to guard the access to the state via another locking mechanism, as this can be set and read by the application thread or the BT RX thread

Reply:
Actually yes: A procedure may be active and aborted. If we are awaiting responses from other devices after one have failed, the procedure is still active in the sense that we cannot start any more, and it is aborted to stop the stack from going forwards with the next step. 

We could introduce a new state, IDLE, and check for that instead of `ACTIVE` if we want to keep this a single value.